Apps/Sandcastle/gallery/Drawing on Terrain.html

```diff
@@ -0,0 +1,185 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
+    <meta name="description" content="Draw lines and polygons on terrain with mouse clicks.">
+    <meta name="cesium-sandcastle-labels" content="Showcases">
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script type="text/javascript" src="../../../ThirdParty/requirejs-2.1.20/require.js"></script>
+    <script type="text/javascript">
+    require.config({
+        baseUrl : '../../../Source',
+        waitSeconds : 60
+    });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+
+    #viewChanged, #cameraChanged {
+        display: none;
+        background-color: red;
+        color: white;
+    }
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer',  {
+    selectionIndicator : false,
+    terrainProvider : Cesium.createWorldTerrain()
+});
+// Create a function to place a point given a world position.
+function createPoint(worldPosition) {
+    var point = new Cesium.Entity({
+        position : worldPosition,
+        point : {
+            color : Cesium.Color.WHITE,
+            pixelSize : 5,
+            heightReference: Cesium.HeightReference.CLAMP_TO_GROUND
+        }
+    });
+    viewer.entities.add(point);
+    return point;
+}
+// On left mouse click, we want to:
+// * Start a new shape if one doesn't already exist.
+// * Add a vertex to the active shape if one exists.
+var activeShapePoints = [];
+var activeShape;
+var floatingPoint;
+var drawingMode = 'line';
+var handler = new Cesium.ScreenSpaceEventHandler(viewer.canvas);
+handler.setInputAction(function(event) {
+    // We use `viewer.scene.pickPosition` here instead of `viewer.camera.pickEllipsoid` so that
+    // we get the correct point when mousing over terrain.
+    var earthPosition = viewer.scene.pickPosition(event.position);
+    // `earthPosition` will be undefined if our mouse is not over the globe.
+    if (Cesium.defined(earthPosition)) {
+        // If this is the first point, we need to initialize the shape.
+        if (activeShapePoints.length === 0) {
```

Comment:
This can be combined with the above if statement.  
 `if (Cesium.defined(earthPosition) && activeShapePoints.length === 0)`

Reply:
There's actually a couple of things that happen even if `activeShapePoints.length != 0` that's why it's 2 statements.